### ISSUE TITLE
Unable to establish connection to Livy endpoint with error 408

### ISSUE THREAD

**Issue by nixent** [microsoft/dbt-fabricspark#19]
Trying to materialize a model with the dbt-fabric adapter. It fails to establish connection to Livy endpoint and times out at [create_session](https://github.com/microsoft/dbt-fabricspark/blob/f2c684e4ea75b675d0e581c121c90458976809f5/dbt/adapters/fabricspark/livysession.py#L131)

Error message:
`Http Error:  408 Client Error: Request Timeout for url: https://api.fabric.microsoft.com/v1/workspaces/{workspace_id}/lakehouses/{lakehouse_id}/livyapi/versions/2023-12-01/sessions`

Call to other endpoints, like [List tables](https://learn.microsoft.com/en-us/rest/api/fabric/lakehouse/tables/list-tables?tabs=HTTP#interface) works fine.

Is there any configuration that needs to be done to make it to work?

**Comment by jcvdodson** [comment]
Are you always getting this error? There shouldn't be any additional config needed.

**Comment by nixent** [comment]
@jcvdodson no, now with same code I'm getting error 404

**Comment by jcvdodson** [comment]
Hmm, that's interesting. Have you double-checked your workspace_id and lakehouse_id in the profiles yml?

**Comment by nixent** [comment]
yes, I have checked connection by calling [List tables](https://learn.microsoft.com/en-us/rest/api/fabric/lakehouse/tables/list-tables?tabs=HTTP#interface) as it includes both workspace_id and lakehouse_id, and it works fine.

I learned about the dbt-fabricspark adapter from dbt channel on Slack where @prdpsvs mentioned that Livy endpoints are in private preview and I should contact our Microsoft account team to sign up for the preview which I have done. Could be that something else needs to be done to activate it for our Fabric capacity.

**Comment by jcvdodson** [comment]
If you are able to create Lakehouses, you should be okay as far as Fabric capacity. Livy might have additional restrictions, but given that Synapse also uses Livy, you should also be good there.

If you'd like, you can email me at t-jdodson@microsoft.com and I can see if I am able to reproduce this issue.

**Comment by sschuetz2bit** [comment]
Hi were you able to fix it?
i also get a 404

**Comment by jcvdodson** [comment]
Have you signed up for the Livy Public Preview?

**Comment by sschuetz2bit** [comment]
Not Yet :)

**Comment by sudo-pradip** [comment]
hello @jcvdodson 
how to signed up for the Livy Public Preview?
cause i tried to find in Azure portal > Subscription > Preview Features and didn't find Livy there,
@sschuetz2bit  fixed for you?

**Comment by simaopedrotoga** [comment]
I'm too unable to establish a connection to Livy endpoint with error 404 by using the dbt-fabric adapter.
Is there a solution for this?
